src/main/java/cz/metacentrum/perun/googlegroupconnector/GoogleGroupsServiceImpl.java

```diff
@@ -800,4 +826,286 @@ private void deleteMember(String groupName, String memberId) throws GoogleGroups
+			return teamDrive;
+		}
+
+		public void setTeamDrive(TeamDrive teamDrive) {
+			this.teamDrive = teamDrive;
+		}
+
+		public List<User> getDriveUsers() {
+			return driveUsers;
+		}
+
+		public void setDriveUsers(List<User> driveUsers) {
+			this.driveUsers = driveUsers;
+		}
+	}
+
+	@Override
+	public List<DriveWithMembers> parseTeamDrivesFile(File teamDriveFile) {
+
+		List<DriveWithMembers> result = new ArrayList<>();
+		FileReader fileReader = null;
+
+		try {
+
+			fileReader = new FileReader(teamDriveFile);
+
+			char separator = ';';
+			CSVReader reader = new CSVReader(fileReader, separator);
+
+			List<String[]> lines = reader.readAll();
+			if (lines != null && !lines.isEmpty()) {
+
+				for (String[] line : lines) {
+
+					if (line.length < 2) {
+						log.error("TeamDrive file contains row with less than 2 columns: {}", line);
+						throw new IllegalArgumentException("TeamDrive file contains row with less than 3 columns:" + line[0]);
+					}
+
+					TeamDrive teamDriveResult = new TeamDrive();
+					List<User> userListResult = new ArrayList<>();
+
+					teamDriveResult.setName(line[0].toLowerCase());
+					List<String> membersEmail = Arrays.asList(line[1].split(","));
+
+					for (String userMail : membersEmail) {
+						userMail = userMail.replaceAll("\\s+", "");
+						User user = new User();
+						user.setPrimaryEmail(userMail);
+						userListResult.add(user);
+					}
+
+					result.add(new DriveWithMembers(teamDriveResult, userListResult));
+				}
+
+				return result;
+			} else {
+				log.error("Team drive file contains no rows.");
+				throw new IllegalArgumentException("Team drive file contains no rows.");
+			}
+		} catch (IOException ex) {
+			log.error("Problem with I/O operation while reading lines of file {} by FileReader.readNext() or getting file: {}", teamDriveFile.getAbsolutePath(), ex);
+		} finally {
+			try {
+				if (fileReader != null) fileReader.close();
+			} catch (IOException ex) {
+				log.error("Problem with I/O operation while closing file {} : {}", teamDriveFile.getAbsolutePath(), ex);
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public void processTeamDrives(List<DriveWithMembers> driveWithMembers) throws GoogleGroupsIOException, InterruptedException {
+
+		TeamDriveList dd = getTeamDrives();
+		List<TeamDrive> domainDrives = new ArrayList<>();
+
+		if (dd != null && !dd.isEmpty() && dd.getTeamDrives() != null) {
+
+			// domain is not empty, compare state
+			domainDrives.addAll(dd.getTeamDrives());
+
+			// create, update team drives
+			for (DriveWithMembers dm : driveWithMembers) {
+				boolean isInPerun = false;
+				for (TeamDrive td : domainDrives) {
+
+					if (td.getName().equals(dm.getTeamDrive().getName())) isInPerun = true;
+
+					if (!Objects.equals(dm.getTeamDrive().getBackgroundImageFile(), td.getBackgroundImageFile()) ||
+							!Objects.equals(dm.getTeamDrive().getColorRgb(), td.getColorRgb()) ||
+							!Objects.equals(dm.getTeamDrive().getThemeId(), td.getThemeId())) {
```

Comment:
Based on docs for ``Drive.create()`` themeId and backgroundFile/colorRgb params can't be set together, but this implementation allows it if changed in a source. But you don't have these properties in an input file, so there is no necessity to update them from Perun. Changing name is not possible from Perun side, since Perun generates new state and old name is lost -> new is created, old deleted. Hence we probably don't need update method.